### ISSUE TITLE
E3002 Additional properties are not allowed ('MaxRecordSizeInKiB' was unexpected)

### ISSUE THREAD

**Issue by jrbarnard** [aws-cloudformation/cfn-lint#4291]
### CloudFormation Lint Version

1.40.0

### What operating system are you using?

Mac

### Describe the bug

E3002 Additional properties are not allowed ('MaxRecordSizeInKiB' was unexpected)

`MaxRecordSizeInKiB` on the `AWS::Kinesis::Stream` resource has just been added to CloudFormation but isn't available in cfn lint yet.

### Expected behavior

It would be available and not raise the E3002 error.

### Reproduction template

```yaml
AWSTemplateFormatVersion: 2010-09-09

Resources:
  Stream:
    Type: AWS::Kinesis::Stream
    Properties:
      StreamModeDetails:
        StreamMode: ON_DEMAND
      RetentionPeriodHours: 336
      MaxRecordSizeInKiB: 10240
      StreamEncryption:
        EncryptionType: KMS
        KeyId: alias/aws/kinesis
```
